### ISSUE TITLE
Invalid term returned by get-value on bvarray term.

### ISSUE THREAD

**Issue by aniemetz** [SRI-CSL/yices2#293]
I get an invalid term when `yices_get_value_as_term` is called on a bvarray that was created from a Bool value and an arithmetic inequality atom. Is this expected behavior?

```
#include <assert.h>                                                             
#include <stdlib.h>                                                             
                                                                                  
#include "yices.h"                                                              
                                                                                  
int                                                                             
main()                                                                          
{                                                                               
  yices_init();                                                                 
  term_t v2 = yices_int64(4101162486);                                          
  term_t c7 = yices_new_uninterpreted_term(yices_bool_type());                  
  term_t c19 = yices_new_uninterpreted_term(yices_int_type());                  
  term_t t44 = yices_idiv(c19, c19);                                            
  term_t t127 = yices_arith_leq_atom(t44, v2);                                  
  term_t args[2] = {c7, t127};                                                  
  term_t t180 = yices_bvarray(2, args);                                         
  ctx_config_t* cfg = yices_new_config();                                       
  context_t* ctx = yices_new_context(cfg);                                      
  assert(yices_check_context(ctx, NULL) == STATUS_SAT);                         
  model_t* model = yices_get_model(ctx, 1);                                                                                                                                                                                  
  term_t v = yices_get_value_as_term(model, t180);                              
  assert(v > 0);                                                                
  return 0;                                                                     
}
```

Yices master 5f3b935e199b8
on Linux

**Comment by disteph** [comment]
Are you sure your assert macro doesn't turn into a noop, in which case there would be no search?

**Comment by BrunoDutertre** [comment]
@aniemetz, there's a division by zero in the call to ``yices_get_value_as_term`` when
computing the value of ``yices_idiv(c19, c19)``.  We should support division by zero in
this case, so that looks like a bug in Yices.

**Comment by BrunoDutertre** [comment]
This should work now. Fixed in  52bbc67e591ca1348a2c5cbadaa515b6875f4d81.

**Comment by aniemetz** [comment]
Works for me.
